NEWS.md

```diff
@@ -4,6 +4,12 @@
 
 * Fixed bug where `step_holiday()` didn't work if it isn't have any missing values. (#1019)
 
+* `step_date()`, `step_dummy()`, `step_dummy_extract()`, `step_holiday()`, `step_ordinalscore()`, and `step_regex()` now returns integer results when appropriate. (#766)
```

Comment:
As an example, `step_dummy()` doesn't return integers for ordered factors. I feel like this bullet would become too hard to read if we have to list all the special cases. 

Essentially, if the output can be an integer, it will be returned as such 😄 


``` r
library(recipes)
library(ggplot2) # For diamonds

recipe(~cut, data = diamonds) %>%
  step_dummy(cut) %>%
  prep() %>%
  bake(new_data = NULL)
#> # A tibble: 53,940 × 4
#>        cut_1  cut_2     cut_3  cut_4
#>        <dbl>  <dbl>     <dbl>  <dbl>
#>  1  6.32e- 1  0.535  3.16e- 1  0.120
#>  2  3.16e- 1 -0.267 -6.32e- 1 -0.478
#>  3 -3.16e- 1 -0.267  6.32e- 1 -0.478
#>  4  3.16e- 1 -0.267 -6.32e- 1 -0.478
#>  5 -3.16e- 1 -0.267  6.32e- 1 -0.478
#>  6 -1.48e-18 -0.535 -3.89e-16  0.717
#>  7 -1.48e-18 -0.535 -3.89e-16  0.717
#>  8 -1.48e-18 -0.535 -3.89e-16  0.717
#>  9 -6.32e- 1  0.535 -3.16e- 1  0.120
#> 10 -1.48e-18 -0.535 -3.89e-16  0.717
#> # … with 53,930 more rows
```

Reply:
Okay, I see! Thanks for clarifying.🙂